### PR TITLE
Fix: "extends" within package.json (fixes #2754)

### LINKED ISSUE #2754: Cannot use "extends" within package.json
Not sure if this is me being stupid or not - but I cannot seem to load the `extends` directive within the package.json `eslintConfig`.

If I have an `.eslintrc` file with the following:

```json
{
    "extends": "strict"
}
```

All works fine. Now if I try to put that in my package.json:

```json
"eslintConfig": {
  "extends": "strict"
}
"devDependencies": {
  "eslint-config-strict": "^1.1.0"
}
```

It reverts to the default eslint configuration - i.e. not extending from the strict config. I couldn't read anything in the docs suggesting this was not allowed.

### PR DESCRIPTION
**Pull request by gyandeeps**
There are many ways to solve this issue. This is my first take on it.
Any suggestions?

### CODE REVIEW

**Comment by nzakas** [comment]
I think there's a much simpler way to handle this, just check if we are reading package.json in `loadConfig()` https://github.com/eslint/eslint/blob/master/lib/config.js#L71

Basically, move https://github.com/eslint/eslint/blob/master/lib/config.js#L223-L230 into `loadConfig()`

**Comment by gyandeeps** [comment]
Thats what I did. But then we basically start considering package.json only as the non-local config file. But my solution right now is open to addition of more non-local config files in the future.

But my first solution was this: https://github.com/eslint/eslint/commit/c87c25e5f65fe726aa133896188eb5eb49b43c80

**Comment by nzakas** [comment]
We aren't going to add any other config file types, so optimizing for that doesn't make sense.

File: lib/config.js [hunk 1]
```diff
@@ -220,13 +238,8 @@ function getLocalConfig(thisConfig, directory) {
...
         localConfig = loadConfig(localConfigFile);
 
-        if (path.basename(localConfigFile) !== LOCAL_CONFIG_FILENAME) {
-
-            // Don't consider a local config file found if the package.json doesn't have the eslintConfig field.
-            if (!localConfig.hasOwnProperty(PACKAGE_CONFIG_FIELD_NAME)) {
-                continue;
-            }
-            localConfig = localConfig[PACKAGE_CONFIG_FIELD_NAME] || {};
+        if (!Object.keys(localConfig).length) {
```
Comment (nzakas) [inline lib/config.js]: Not sure I understand the point of this. An empty config should have no effect, so if just let it through.

**Comment by nzakas** [comment]
Ooh I see now, I got a little confused by the extra function. I just don't see us ever adding more config files, so direct comparison is best.

**Comment by gyandeeps** [comment]
sounds good. I will bring back my initial logic from here : https://github.com/eslint/eslint/commit/c87c25e5f65fe726aa133896188eb5eb49b43c80

File: lib/config.js (hunk 1 above)
Comment (gyandeeps) [inline lib/config.js]: we can't as it will make the found flag true.

Comment (nzakas) [inline lib/config.js]: Oh wait, this is the replacement for lines 226-227.  Can you add a comment? Otherwise this is really unclear.

**Comment by ilyavolodin** [comment]
LGTM

**Pull request merged**